src/containers/BookList.js

```diff
@@ -20,15 +25,31 @@ const BookList = props => {
             <th>Remove Book</th>
 
           </tr>
-          { books.map(elem => (
-            <Book
-              key={elem.id}
-              title={elem.title}
-              category={elem.category}
-              id={elem.id}
-              handleRemoveBook={removeBook}
-            />
-          ))}
+          { books.map(elem => {
+            if (filter === 'All') {
+              return (
+                <Book
+                  key={elem.id}
+                  title={elem.title}
+                  category={elem.category}
+                  id={elem.id}
+                  handleRemoveBook={removeBook}
+                />
+              );
+            }
+            if (elem.category === filter) {
+              return (
+                <Book
+                  key={elem.id}
+                  title={elem.title}
+                  category={elem.category}
+                  id={elem.id}
+                  handleRemoveBook={removeBook}
+                />
+              );
+            }
+            return undefined;
```

Comment:
this is a presentation component it doesn't need to return this

Reply:
thank you!